src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/security/CaseloadAccess.kt

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.security
+
+import org.springframework.web.context.request.RequestContextHolder
+import org.springframework.web.context.request.ServletRequestAttributes
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.getBvlsRequestContext
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.UserType
+
+/**
+ * Caseload access checks are only relevant for prison users. If the current user is not a prison user then it will just
+ * be ignored.
+ */
+fun checkCaseLoadAccess(prisonCode: String) {
+  val httpRequest = if (RequestContextHolder.getRequestAttributes() != null) (RequestContextHolder.getRequestAttributes() as ServletRequestAttributes).request else throw CaseloadAccessException()
```

Comment:
This line probably could throw an alternative runtime exception if the expected http attrs are null.  CaseloadAccessException is possibly misleading???